registry/storage/driver/s3-aws/s3.go

```diff
@@ -1300,7 +1357,7 @@ func (w *writer) Write(p []byte) (int, error) {
 	// If the last written part is smaller than minChunkSize, we need to make a
 	// new multipart upload :sadface:
 	if len(w.parts) > 0 && int(*w.parts[len(w.parts)-1].Size) < minChunkSize {
-		var completedUploadedParts completedParts
+		completedUploadedParts := make(completedParts, 0, len(w.parts))
 		for _, part := range w.parts {
 			completedUploadedParts = append(completedUploadedParts, &s3.CompletedPart{
```

Comment:
```suggestion
		completedUploadedParts := make(completedParts, len(w.parts))
		for i, part := range w.parts {
			completedUploadedParts[i] = &s3.CompletedPart{
```
for consistency with lines 1546-8